src/hotspot/cpu/x86/vm_version_x86.cpp

```diff
@@ -898,6 +898,16 @@ void VM_Version::get_processor_features() {
     FLAG_SET_DEFAULT(UseCRC32Intrinsics, false);
   }
 
+  if (supports_avx2() && UseAdler32Intrinsics) {
```

Comment:
This should be under #ifdef _LP64. 
For 32-bit, UseAdler32Intrinsics should be set to false.